pkg/controller/pxcrestore/util.go

```diff
@@ -95,3 +98,34 @@ func isJobFinished(checkJob *batchv1.Job) (bool, error) {
 	}
 	return false, nil
 }
+
+func (r *ReconcilePerconaXtraDBClusterRestore) isPITRReady(ctx context.Context, cluster *api.PerconaXtraDBCluster, bcp *api.PerconaXtraDBClusterBackup) (bool, error) {
+	cond := meta.FindStatusCondition(bcp.Status.Conditions, api.BackupConditionPITRReady)
+	if cond != nil && cond.Status == metav1.ConditionFalse {
+		return false, nil
+	}
+
+	opts, err := storage.GetOptionsFromBackup(ctx, r.client, cluster, bcp)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get storage options")
+	}
+
+	stg, err := r.newStorageClientFunc(ctx, opts)
+	if err != nil {
+		return false, errors.Wrap(err, "new storage")
+	}
+	found := true
+
+	filepath := bcp.Status.Destination.BackupName() + "." + naming.PITRNotReady
+	objReader, err := stg.GetObject(ctx, filepath)
+	if err != nil {
+		if !errors.Is(err, storage.ErrObjectNotFound) {
+			return false, errors.Wrap(err, "get pitr-not-ready file from storage")
+		}
+		found = false
+	} else {
+		objReader.Close()
+	}
+
+	return !found, nil
+}
```

Comment:
feel free the reject but i think this version can be more readable and we can get rid of `found` variable

```suggestion
	if err == nil || errors.Is(err, storage.ErrObjectNotFound) {
	    objReader.Close()
		return false, nil
	}
	
	return false, errors.Wrap(err, "get pitr-not-ready file from storage")
```